src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -166,34 +166,43 @@ protected override void OnAfterRender(bool firstRender)
         }
     }
 
-    private void UpdateResourcesList()
+    internal static ImmutableList<SelectViewModel<ResourceTypeDetails>> GetConsoleLogResourceSelectViewModels(
+        ConcurrentDictionary<string, ResourceViewModel> resourcesByName,
+        SelectViewModel<ResourceTypeDetails> noSelectionViewModel,
+        string resourceUnknownStateText)
     {
         var builder = ImmutableList.CreateBuilder<SelectViewModel<ResourceTypeDetails>>();
 
-        foreach (var resourceGroupsByApplicationName in _resourceByName
+        foreach (var grouping in resourcesByName
             .Where(r => !r.Value.IsHiddenState())
-            .OrderBy(c => c.Value.Name)
-            .GroupBy(r => r.Value.DisplayName, r => r.Value))
+            .OrderBy(c => c.Value.Name, StringComparers.ResourceName)
+            .GroupBy(r => r.Value.GetReplicaSetOrDefault()?.Uid ?? r.Value.Uid))
         {
-            if (resourceGroupsByApplicationName.Count() > 1)
+            string applicationName;
+
+            if (grouping.Count() > 1)
             {
+                applicationName = grouping.First().Value.GetReplicaSetOrDefault()!.Name;
```

Comment:
What is the situation where the replica set name is different from DisplayName?

Reply:
For context, I was told (a while ago) that we should not assume that replica sets can be derived from the resource name and that we should at some point use info from dcp. Wondering if that’s changed, would love Karol’s thoughts

Reply:
We did in fact make a change (recently, https://github.com/microsoft/usvc-apiserver/pull/410) that made it official that names of the Executables belonging to a replica set are derived from the replica set by applying a unique suffix. 

As for `DisplayName`, I could not find any place in Aspire code that sets this property other than https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting/Dashboard/DashboardServiceData.cs#L39  If I am right, then DisplayName and resource name are the same.

FWIIW, inside DCP we do generate a "replica display name" as an annotation on an Executable. It is built off of replica set name and monotonically increasing counter. At some point we thought that might be something the dashboard could leverage, but it never happened.

Reply:
> We did in fact make a change (recently, https://github.com/microsoft/usvc-apiserver/pull/410) that made it official that names of the Executables belonging to a replica set are derived from the replica set by applying a unique suffix.

So, the owner information part can be removed then? 🙂

Reply:
You do not have to scan the owner objects (although there is no harm in that), but please do not parse the name of the Executable either. Instead, if you want to know the replica set name, there is now an annotation for that `executable-replica-set.usvc-dev.developer.microsoft.com/replica-set-name` This will be put on every Executable launched by a replica set.

Caveat though: this is added by DCP 0.5.6 or newer; DCP 0.2.3 (Aspire 8.0 version) does not have this. So scanning owners may be your best bet, at least short-term, if backward compat is a concern

Reply:
In that case, it might be better to scan owners? And then if there are additional owners we can show in the future on the dashboard, it will be easier?

Reply:
That would be my recommendation, yes.

Reply:
I still don't understand what bug is actually being fixed here. What is broken today that this fixes? How do I repo what is broken today so I can see it for myself?

Reply:
There is no broken behavior, but per @karolz-ms's comment above,
> please do not parse the name of the Executable either

is what we're doing already with the `DisplayName` https://github.com/dotnet/aspire/pull/4908/files#diff-e2cbd8471a0c18b0125d89b07d3d720c283699131220c4f573967e138f7ff4cb

The main purpose of this PR is to add unit tests for this behavior (#1411), but I took the opportunity to change `GetResourceName` to use the actual replica set information.

Reply:
We're not parsing the name. The linked change doesn't show anything that parses the name.

As far as I can tell the DisplayName we have on the resource is already the replica set name. This seems like more complicated way of getting the same result we already have.

Reply:
@karolz-ms could you weigh in as well with your thoughts?

Reply:
Can one of you point me to the current implementation that sets the `DisplayName` property on Executables that belong to the replica set? The one that sets it to replica set name?

Reply:
I believe it comes from the `resource-name` annotation.

![image](https://github.com/user-attachments/assets/2a5105d6-6444-4892-b791-d45e8351f140)

Reply:
If that is the case, the DisplayName is not necessarily the name of the Executable replica set. The `resource-name` is an annotation that ApplicationExecutor class seems to be applying; it was introduced by https://github.com/dotnet/aspire/pull/3103. Compare line https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L1012 and line https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting/Dcp/ApplicationExecutor.cs#L1019 and notice that the Executable replica set name can have suffixes that `resouce-name` annotation will not have. 

If the goal here is to group Executables by its owning replica set (including the group for Executables that have no owning replica set, i.e. standalone) then what @adamint has seems sound to me.

Reply:
If my understanding is correct, the difference is the `resource-name` could be missing `DcpOptions.ResourceNameSuffix`. What is that for, and when it would be set?

I tried configuring in a host JSON config file:

```json
{
  "DcpPublisher": {
    "ResourceNameSuffix": "r"
  }
}
```

And I found that setting that didn't impact grouping values in the UI:

![image](https://github.com/user-attachments/assets/edfa8873-cb5b-4843-aa55-e42fe46852be)

The only oddity I see is the grouped name is `catalogservice` and not `catalogservice-r`. Does that matter?

Reply:
> If my understanding is correct, the difference is the `resource-name` could be missing `DcpOptions.ResourceNameSuffix`. What is that for, and when it would be set?
> 
The suffix is used by tests to ensure that object have probabilistically unique names, e.g. https://github.com/dotnet/aspire/blob/main/src/Aspire.Hosting.Testing/DistributedApplicationFactory.cs#L163 

> 
> The only oddity I see is the grouped name is `catalogservice` and not `catalogservice-r`. Does that matter? Is it desirable? The display name is `catalogservice` everywhere else.

That is for you to decide, not me ☺️ I am just trying to help by answering questions about what is guaranteed/safe in terms of resource naming and DCP orchestration.

Reply:
In that case, resource name and replica set name will always be the same in real world usage.

Reply:
Sounds good, I'll remove the owner changes then. @JamesNK